testutil/validatormock/synccomm.go

```diff
@@ -316,3 +325,72 @@ func submitSyncMessage(ctx context.Context, eth2Cl eth2wrap.Client, slot eth2p0.
 
 	return nil
 }
+
+// aggContributions submits aggregate altair.SignedContributionAndProof. It returns false if contribution aggregation is not required.
+func aggContributions(ctx context.Context, eth2Cl eth2wrap.Client, signFunc SignFunc, slot eth2p0.Slot, vals validators, selections syncSelections, blockRoot *eth2p0.Root) (bool, error) {
+	if len(selections) == 0 {
+		return false, nil
+	}
+
+	epoch, err := eth2util.EpochFromSlot(ctx, eth2Cl, slot)
+	if err != nil {
+		return false, err
+	}
+
+	var signedContribAndProofs []*altair.SignedContributionAndProof
+	for _, selection := range selections {
+		// Check if the validator is an aggregator.
+		ok, err := eth2exp.IsSyncCommAggregator(ctx, eth2Cl, selection.SelectionProof)
```

Comment:
rather follow SlotAttester pattern where `IsAttAggregator` is called in the prepare/selection step, so only cache selections that are actually aggregators

Reply:
because you will return true below even when no validators are aggregators from this slot. And `SubmitSyncCommitteeContributions` will probably error if no validators are aggregators for this slot.